### PR TITLE
Fixes #21064 - Fix scenario context in hooks

### PR DESCRIPTION
**Pull request by ekohl**

### CODE REVIEW

File: test/kafo/hook_context_test.rb [hunk 1]
```diff
@@ -11,6 +11,24 @@ module Kafo
...
+      specify { context.respond_to?(:get_custom_config).must_equal true }
+      specify { context.respond_to?(:store_custom_config).must_equal true }
+      specify { context.respond_to?(:scenario_path).must_equal true }
+      specify { context.respond_to?(:scenario_data).must_equal true }
+    end
+
+    describe "#scenario_data" do
+      specify do
+        Tempfile.open('scenario') do |file|
+          file.write(YAML.dump({foo: 'bar'}))
```
Comment (ares) [inline test/kafo/hook_context_test.rb]: this wouldn't work with Ruby 1.8 which is still used with puppet 3.8, would you mind using hash rocket `=>` syntax here please?

File: test/kafo/hook_context_test.rb [hunk 2]
```diff
@@ -11,6 +11,24 @@ module Kafo
...
+          file.flush
+
+          context.stub :scenario_path, file.path do
+            assert_equal context.scenario_data, {foo: 'bar'}
```
Comment (ares) [inline test/kafo/hook_context_test.rb]: same as above

File: test/kafo/hook_context_test.rb (hunk 1 above)
Comment (ekohl) [reply test/kafo/hook_context_test.rb]: Updated

**Comment by ares** [comment]
Thanks @ekohl, merging after running tests locally.

**Pull request merged**